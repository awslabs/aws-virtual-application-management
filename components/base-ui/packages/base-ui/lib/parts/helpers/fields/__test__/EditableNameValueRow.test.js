/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import EditableNameValueRow from '../EditableNameValueRow';

describe('EditableNameValueRow', () => {
  it('renders correctly', () => expect(renderer.create(<EditableNameValueRow />).toJSON()).toMatchSnapshot());

  it('handles submit correctly', async () => {
    const mockEvent = { preventDefault: jest.fn(), stopPropagation: jest.fn() };
    const form = { errors: () => [] };
    const onExitEditMode = jest.fn();
    await shallow(<EditableNameValueRow form={form} onExitEditMode={onExitEditMode} />)
      .find('EditableField')
      .simulate('submit', mockEvent);
    expect(onExitEditMode).toHaveBeenCalledTimes(1);
  });
});
