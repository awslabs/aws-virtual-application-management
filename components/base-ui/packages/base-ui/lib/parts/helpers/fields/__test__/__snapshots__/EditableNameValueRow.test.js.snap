// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableNameValueRow renders correctly 1`] = `
<tr
  className=""
>
  <td
    className=""
  />
  <td
    className=""
  />
  <td
    className=""
  >
    <i
      aria-hidden="true"
      className="grey pencil icon ml1 cursor-pointer"
      onClick={[Function]}
    />
    <i
      aria-hidden="true"
      className="grey trash icon ml1 cursor-pointer"
      onClick={[Function]}
    />
  </td>
</tr>
`;
